llama_stack/providers/utils/tools/mcp.py

```diff
@@ -43,9 +47,46 @@ async def sse_client_wrapper(endpoint: str, headers: dict[str, str]):
         raise
 
 
+@asynccontextmanager
+async def streamable_http_client_wrapper(endpoint: str, headers: dict[str, str]) -> AsyncGenerator[ClientSession, Any]:
+    try:
+        async with streamablehttp_client(endpoint, headers=headers) as (
+            read_stream,
+            write_stream,
+            _,
+        ):
+            async with ClientSession(read_stream, write_stream) as session:
+                await session.initialize()
+                yield session
+    except* httpx.HTTPStatusError as eg:
+        for exc in eg.exceptions:
+            # mypy does not currently narrow the type of `eg.exceptions` based on the `except*` filter,
+            # so we explicitly cast each item to httpx.HTTPStatusError. This is safe because
+            # `except* httpx.HTTPStatusError` guarantees all exceptions in `eg.exceptions` are of that type.
+            err = cast(httpx.HTTPStatusError, exc)
+            if err.response.status_code == 401:
+                raise AuthenticationRequiredError(exc) from exc
+        raise
+
+
+def get_client_wrapper(endpoint: str, headers: dict[str, str]) -> _AsyncGeneratorContextManager[ClientSession, Any]:
+    path = PurePosixPath(unquote(urlparse(endpoint).path))
+    if path.parts[-1] != "sse":
```

Comment:
```suggestion
    if path.parts and path.parts[-1] != "sse":
```

Reply:
To avoid an out of range if sse does not exist?

Reply:
Given the conversation above, I removed the optimization entirely, as it seems like the consensus is that this was not providing much value